libc/test/src/math/smoke/SetPayloadTest.h

```diff
@@ -57,7 +62,15 @@ class SetPayloadTestTemplate : public LIBC_NAMESPACE::testing::FEnvSafeTest {
     EXPECT_EQ(FPBits::quiet_nan(Sign::POS, 0x123).uintval(),
               FPBits(res).uintval());
 
-    EXPECT_EQ(0, func(&res, T(FPBits::FRACTION_MASK >> 1)));
+    // The following code is creating a NaN manually to prevent a conversion
```

Comment:
Nit: it's creating the payload which will be stored in the NaN mantissa. It's not creating the NaN itself.